added sql reference (#4746)
influxdata/docs-v2: content/influxdb/cloud-iox/reference/sql/_index.md

```

### Selector functions

Selector functions are unique to InfluxDB. They behave like aggregate functions in that they take a row of data and compute it down to a single value.  However, selectors are unique in that they return a **time value** in addition to the computed value. In short, selectors retrun an aggreagetd value along with a timestamp. 

| Function         | Description                                                     |
| :--------------- | :-------------------------------------------------------------- |
| SELECTOR_FIRST() | Returns the first value of a selected column and timestamp.   |
| SELECTOR_LAST()  | Returns the last value of a selected column and timestamp.     |
| SELECTOR_MIN()   | Returns the smallest value of a selected column and timestamp. |
| SELECTOR_MAX()   | Returns the largest value of a selected column and timestamp.  |

#### Examples

```sql
SELECT 
SELECTOR_MAX("pH", time)['value'],
SELECTOR_MAX("pH", time)['time']
FROM "h2o_pH"

SELECT 
SELECTOR_LAST("water_level", time)['value'],
SELECTOR_LAST("water_level", time)['time']
FROM "h2o_feet"
WHERE time >= timestamp '2019-09-10T00:00:00Z' AND time <= timestamp '2019-09-19T00:00:00Z'

```

> ### Time series functions

Reviewer: These are necessarily unique to "time series."

```suggestion
### Date and time functions
```

| Function     | Description                                                                                     |
| :----------- | :---------------------------------------------------------------------------------------------- |
| DATE_BIN()   | Bins the input timestamp into a specified interval.                                             |
| DATE_TRUNC() | Truncates a timestamp expression based on the date part specified, such as hour, day, or month. |
| DATE_PART()  | Returns the specified part of a date.                                                           |

> | NOW()        | Returns the current time.                                                                       |

Reviewer: ```suggestion
| NOW()        | Returns the current time (UTC).                                                                  |
```

                 
#### Examples

```sql
SELECT DATE_BIN(INTERVAL '1 hour', time, '2019-09-18T00:00:00Z'::timestamp),
SUM(water_level)
FROM "h2o_feet"
GROUP BY time

SELECT DATE_TRUNC('month',time) AS "date",
SUM(water_level)
FROM "h2o_feet"
GROUP BY time
```

### Approximate functions

| Function      | Description                                    |
| :------------ | :--------------------------------------------- |

> | APPROX_MEDIAN | Returns the approximate median of input values. |

Reviewer: This is missing functions.

- approx_distinct
- approx_percentile_cont
- approx_percentile_cont_with_weight

Author: adding now

'2022-01-31 06:30:30.123'      -- (RFC3339-like)
'2022-01-31 06:30:30'          -- ((RFC3339-like, no fractional seconds) 
1643610630123000000::TIMESTAMP -- (Unix epoch nanosecond cast to a timestamp)
```

### Boolean literals

Boolean literals are either `TRUE` or `FALSE`. 

## Duration units

Interval literals specify a length or unit of time. 

```sql
INTERVAL '4 minutes'
INTERVAL '12 days 6 hours 30 minutes'
```

## Operators

Operators are reserved words or characters which perform certain operations, inluding comparisons and arithmetic. 

### Arithmetic operators

Arithmetic operators take two numeric values (either literals or variables) and
perform a calculation that returns a single numeric value.

| Operator | Description    | Example | Result |
| :------: | :------------- | ------- | -----: |
|   `+`    | Addition       | `2 + 2` |    `4` |
|   `-`    | Subtraction    | `4 - 2` |    `2` |
|   `*`    | Multiplication | `2 * 3` |    `6` |
|   `/`    | Division       | `6 / 3` |    `2` |
|   `%`    | Modulo         | `7 % 2` |    `1` |

### Comparison operators

Comparison operators evaluate the relationship between the left and right operands and `TRUE` or `FALSE`.

> | Operator | Meaning                               | Example          |
> | :------: | :------------------------------------ | :--------------- |
> |   `=`    | Equal to                              | `123 = 123`      |
> |   `<>`   | Not equal to                          | `123 <> 456`     |
> |   `!=`   | Not equal to                          | `123 != 456`     |
> |   `>`    | Greater than                          | `3 > 2`          |
> |   `>=`   | Greater than or equal to              | `3 >= 2`         |
> |   `<`    | Less than                             | `1 < 2`          |
> |   `<=`   | Less than or equal to                 | `1 <= 2`         |
> |   `~`    | Matches a regular expression          | `'abc' ~ 'a.*'`  |
> |   `!~`   | Does not matches a regular expression | `'abc' !~ 'd.*'` |

Reviewer: ```suggestion
| Operator | Meaning                             | Example          |
| :------: | :---------------------------------- | :--------------- |
|   `=`    | Equal to                            | `123 = 123`      |
|   `<>`   | Not equal to                        | `123 <> 456`     |
|   `!=`   | Not equal to                        | `123 != 456`     |
|   `>`    | Greater than                        | `3 > 2`          |
|   `>=`   | Greater than or equal to            | `3 >= 2`         |
|   `<`    | Less than                           | `1 < 2`          |
|   `<=`   | Less than or equal to               | `1 <= 2`         |
|   `~`    | Matches a regular expression        | `'abc' ~ 'a.*'`  |
|   `!~`   | Does not match a regular expression | `'abc' !~ 'd.*'` |
```

## Keywords

> The following reserved key words cannot be used as identifiers.

Reviewer: ```suggestion
The following reserved keywords cannot be used as identifiers.
```
